src/java.base/share/classes/java/nio/charset/Charset.java

```diff
@@ -592,11 +591,18 @@ private static void put(Iterator<Charset> i, Map<String,Charset> m) {
     /**
      * Returns the default charset of this Java virtual machine.
      *
-     * <p> The default charset is determined during virtual-machine startup and
-     * typically depends upon the locale and charset of the underlying
-     * operating system.
+     * <p> The default charset is {@code UTF-8}, unless changed in an
+     * implementation specific manner.
+     *
+     * @implNote An implementation may override the default charset with
+     * the system property {@code file.encoding} on the command line. If the
+     * value designates {@code COMPAT}, the default charset is derived from
```

Comment:
It might be simpler to say "If the value is COMPAT" and avoid the word "designates" here.